python/cugraph/structure/renumber.py

```diff
@@ -63,3 +66,105 @@ def renumber(source_col, dest_col):
                                                                  dest_col)
 
     return source_col, dest_col, numbering_map
+
+
+def renumber_from_cudf(_df, source_cols_names, dest_cols_names):
+    """
+    Take a set, collection (lists) of source and destination columns, and
+    renumber the vertices to create a dense set of contiguously vertex ids
+    from 0 to the number of unique vertices - 1.
+
+    Input columns can be any data type.
+
+    The output will be mapped to int32, since many of the cugraph functions
+    are limited to int32. If the number of unique values is > 2^31-1 then
+    this function will return an error.
+
+    NOTICE
+    ---------
+    - The number of source and destination columns must be the same
+    - The source and destination column names cannot be the same or overlap.
+    - The data type order needs to be the same between source and destination
+        columns. This is due to the two sets being merged to create a single
+        list of all possible values
+
+    Input Parameters
+    ----------
+    df : cudf.DataFrame
+        The dataframe containing the source and destination columans
+    source_cols_names : List
+        This is a list of source column names
+    dest_cols_names : List
+        This is a list of destination column names
+
+    Returns
+    ---------
+    src_ids : cudf.Series
```

Comment:
A Dataframe instead of 2 Series would be ideal since this is what we accept in `from_cudf_edgelist`

Reply:
I need to do some performance testing before we drop the old version.  There is a cudf merge issues where it does not guarantee order, so there are two extra sorts that might need to be improved